packages/react-reconciler/src/ReactFiberCommitWork.old.js

```diff
@@ -2414,24 +2416,54 @@ function getRetryCache(finishedWork) {
 }
 
 export function detachOffscreenInstance(instance: OffscreenInstance): void {
-  const currentOffscreenFiber = instance._current;
-  if (currentOffscreenFiber === null) {
+  const fiber = instance._current;
+  if (fiber === null) {
     throw new Error(
       'Calling Offscreen.detach before instance handle has been set.',
     );
   }
 
-  const executionContext = getExecutionContext();
-  if ((executionContext & (RenderContext | CommitContext)) !== NoContext) {
-    scheduleMicrotask(() => {
-      instance._visibility |= OffscreenDetached;
-      disappearLayoutEffects(currentOffscreenFiber);
-      disconnectPassiveEffect(currentOffscreenFiber);
-    });
-  } else {
+  if ((instance._visibility & OffscreenDetached) !== NoFlags) {
+    // The instance is already detached, this is a noop.
+    return;
+  }
+
+  instance._pendingVisibility |= OffscreenDetached;
+
+  // Detaching needs to be postoned in case attach is called before next update.
+  scheduleMicrotask(() => {
```

Comment:
Since all this does now is schedule an update, you don't need the `scheduleMicrotask` call

Reply:
I was using scheduleMicrotask to postpone detachment of Offscreen if `detach` is called in an effect or in render phase. I'm not sure how to do that with `scheduleUpdateOnFiber`.